### ISSUE TITLE
Can't compile for flash

### ISSUE THREAD

**Issue by RudolfVonKrugstein** [openfl/lime#345]
Trying to compile an openfl sample like this:

```
> lime create openfl:PiratePig
> lime test flash
Error: SWF Library not found : /home/ls/src/PiratePig/Export/flash/obj/assets.swf
```

**Comment by jgranick** [comment]
That's strange, could you try a -verbose build? Thank you! :smile:

**Comment by RudolfVonKrugstein** [comment]
of course:

```
> lime build flash -verbose
 ___     
/\_ \    __                       
\//\ \  /\_\    ___ ___      __   
  \ \ \ \/\ \ /' __` __`\  /'__`\ 
   \_\ \_\ \ \/\ \/\ \/\ \/\  __/ 
   /\____\\ \_\ \_\ \_\ \_\ \____\
   \/____/ \/_/\/_/\/_/\/_/\/____/

Lime Command-Line Tools (2.1.2)

Initializing project...
Using project file: /home/ls/src/PiratePig/project.xml
Reading HXCPP config: /home/ls/.hxcpp_config.xml
Using target platform: FLASH

Running command: UPDATE
 - Embedding asset: (FONT) Assets/fonts/FreebooterUpdated.ttf
 - Embedding asset: (IMAGE) Assets/images/game_bunny_02.png
 - Embedding asset: (IMAGE) Assets/images/game_carrot.png
 - Embedding asset: (IMAGE) Assets/images/center_bottom.png
 - Embedding asset: (IMAGE) Assets/images/background_tile.png
 - Embedding asset: (IMAGE) Assets/images/game_lemon.png
 - Embedding asset: (IMAGE) Assets/images/game_piratePig.png
 - Embedding asset: (IMAGE) Assets/images/game_bear.png
 - Embedding asset: (IMAGE) Assets/images/game_panda.png
 - Embedding asset: (IMAGE) Assets/images/logo.png
 - Embedding asset: (SOUND) Assets/sounds/3.mp3
 - Embedding asset: (SOUND) Assets/sounds/4.mp3
 - Embedding asset: (SOUND) Assets/sounds/5.mp3
 - Embedding asset: (MUSIC) Assets/sounds/theme.mp3
 - Embedding asset: (FONT) Assets/fonts/FreebooterUpdated.ttf
 - Embedding asset: (IMAGE) Assets/images/game_bunny_02.png
 - Embedding asset: (IMAGE) Assets/images/game_carrot.png
 - Embedding asset: (IMAGE) Assets/images/center_bottom.png
 - Embedding asset: (IMAGE) Assets/images/background_tile.png
 - Embedding asset: (IMAGE) Assets/images/game_lemon.png
 - Embedding asset: (IMAGE) Assets/images/game_piratePig.png
 - Embedding asset: (IMAGE) Assets/images/game_bear.png
 - Embedding asset: (IMAGE) Assets/images/game_panda.png
 - Embedding asset: (IMAGE) Assets/images/logo.png
 - Embedding asset: (SOUND) Assets/sounds/3.mp3
 - Embedding asset: (SOUND) Assets/sounds/4.mp3
 - Embedding asset: (SOUND) Assets/sounds/5.mp3
 - Embedding asset: (MUSIC) Assets/sounds/theme.mp3
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/haxe/DefaultAssetLibrary.hx -> Export/flash/haxe/DefaultAssetLibrary.hx
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/haxe/ApplicationMain.hx -> Export/flash/haxe/ApplicationMain.hx
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/haxe/ApplicationMain.hx -> Export/flash/haxe/ApplicationMain.hx
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/haxe/NMEPreloader.hx -> Export/flash/haxe/NMEPreloader.hx
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/flash/hxml/final.hxml -> Export/flash/haxe/final.hxml
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/flash/hxml/debug.hxml -> Export/flash/haxe/debug.hxml
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/flash/hxml/release.hxml -> Export/flash/haxe/release.hxml
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/flash/hxml/final.hxml -> Export/flash/haxe/final.hxml
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/flash/hxml/debug.hxml -> Export/flash/haxe/debug.hxml
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/flash/hxml/release.hxml -> Export/flash/haxe/release.hxml

Running command: BUILD
 - Running command: haxe -main ApplicationMain -cp /usr/lib/haxe/lib/openfl/2,2,7 -D openfl=2.2.7 -cp /usr/lib/haxe/lib/lime/2,1,2 -D lime=2.1.2 -cp /usr/lib/haxe/lib/actuate/git -D actuate=1.8.2 -cp Source -D lime-openal -D tools=2.1.2 -D flash-use-stage -D no-compilation -D lime-opengl -D openfl-flash -D lime-curl -D web -swf-lib /home/ls/src/PiratePig/Export/flash/obj/assets.swf -swf-version 11.2 -swf Export/flash/bin/PiratePig.swf -cp Export/flash/haxe
Error: SWF Library not found : /home/ls/src/PiratePig/Export/flash/obj/assets.swf
```

**Comment by RudolfVonKrugstein** [comment]
By the way, /home/ls/src/PiratePig/Export/flash/obj/assets.swf does exist (at least after running the command).

**Comment by RudolfVonKrugstein** [comment]
With 2.0.6 it works, and the notable difference is, that 2.0.6 uses a relative path for the assets.swf:

```
-swf-lib Export/flash/obj/assets.swf
```

**Comment by jgranick** [comment]
Oh, interesting. I got a pull request from someone, that said that having two swf libs of the same name was causing a conflict, so the absolute path resolved that

It's been working here with absolute paths, here, what OS and version are you using, so I can try and test it?

**Comment by RudolfVonKrugstein** [comment]
I am on ubuntu linux 14.04 64 bit:

```
3.16.0-30-generic #40-Ubuntu SMP Mon Jan 12 22:06:37 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by jgranick** [comment]
Hmm, this is strange. I just installed the latest on the server (which currently runs Ubuntu 12.04), here's what I get:

```bash
joshua@ponyo:~/Projects/Samples/PiratePig$ lime build flash -v
 ___
/\_ \    __
\//\ \  /\_\    ___ ___      __
  \ \ \ \/\ \ /' __` __`\  /'__`\
   \_\ \_\ \ \/\ \/\ \/\ \/\  __/
   /\____\\ \_\ \_\ \_\ \_\ \____\
   \/____/ \/_/\/_/\/_/\/_/\/____/

Lime Command-Line Tools (2.1.2)

Initializing project...
Using project file: /home/joshua/Projects/Samples/PiratePig/project.xml
Reading HXCPP config: /home/joshua/.hxcpp_config.xml
Using target platform: FLASH

Running command: UPDATE
 - Embedding asset: (FONT) Assets/fonts/FreebooterUpdated.ttf
 - Embedding asset: (IMAGE) Assets/images/game_bear.png
 - Embedding asset: (IMAGE) Assets/images/game_piratePig.png
 - Embedding asset: (IMAGE) Assets/images/background_tile.png
 - Embedding asset: (IMAGE) Assets/images/game_bunny_02.png
 - Embedding asset: (IMAGE) Assets/images/game_lemon.png
 - Embedding asset: (IMAGE) Assets/images/logo.png
 - Embedding asset: (IMAGE) Assets/images/game_panda.png
 - Embedding asset: (IMAGE) Assets/images/center_bottom.png
 - Embedding asset: (IMAGE) Assets/images/game_carrot.png
 - Embedding asset: (SOUND) Assets/sounds/3.mp3
 - Embedding asset: (SOUND) Assets/sounds/4.mp3
 - Embedding asset: (SOUND) Assets/sounds/5.mp3
 - Embedding asset: (MUSIC) Assets/sounds/theme.mp3
 - Embedding asset: (FONT) Assets/fonts/FreebooterUpdated.ttf
 - Embedding asset: (IMAGE) Assets/images/game_bear.png
 - Embedding asset: (IMAGE) Assets/images/game_piratePig.png
 - Embedding asset: (IMAGE) Assets/images/background_tile.png
 - Embedding asset: (IMAGE) Assets/images/game_bunny_02.png
 - Embedding asset: (IMAGE) Assets/images/game_lemon.png
 - Embedding asset: (IMAGE) Assets/images/logo.png
 - Embedding asset: (IMAGE) Assets/images/game_panda.png
 - Embedding asset: (IMAGE) Assets/images/center_bottom.png
 - Embedding asset: (IMAGE) Assets/images/game_carrot.png
 - Embedding asset: (SOUND) Assets/sounds/3.mp3
 - Embedding asset: (SOUND) Assets/sounds/4.mp3
 - Embedding asset: (SOUND) Assets/sounds/5.mp3
 - Embedding asset: (MUSIC) Assets/sounds/theme.mp3
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/haxe/DefaultAss
etLibrary.hx -> Export/flash/haxe/DefaultAssetLibrary.hx
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/haxe/Applicatio
nMain.hx -> Export/flash/haxe/ApplicationMain.hx
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/haxe/NMEPrelo
ader.hx -> Export/flash/haxe/NMEPreloader.hx
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/haxe/Applicat
ionMain.hx -> Export/flash/haxe/ApplicationMain.hx
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/flash/hxml/fina
l.hxml -> Export/flash/haxe/final.hxml
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/flash/hxml/debu
g.hxml -> Export/flash/haxe/debug.hxml
 - Copying template file: /usr/lib/haxe/lib/lime/2,1,2/templates/flash/hxml/rele
ase.hxml -> Export/flash/haxe/release.hxml
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/flash/hxml/fi
nal.hxml -> Export/flash/haxe/final.hxml
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/flash/hxml/de
bug.hxml -> Export/flash/haxe/debug.hxml
 - Copying template file: /usr/lib/haxe/lib/openfl/2,2,7/templates/flash/hxml/re
lease.hxml -> Export/flash/haxe/release.hxml

Running command: BUILD
 - Running command: haxe -main ApplicationMain -cp /usr/lib/haxe/lib/openfl/2,2,7 -D openfl=2.2.7 -cp /usr/lib/haxe/lib/lime/2,1,2 -D lime=2.1.2 -cp /usr/lib/haxe/lib/actuate/1,8,2 -D actuate=1.8.2 -cp Source -D lime-openal -D tools=2.1.2 -D flash-use-stage -D no-compilation -D lime-opengl -D openfl-flash -D lime-curl -D web -swf-lib /home/joshua/Projects/Samples/PiratePig/Export/flash/obj/assets.swf -swf-version 11.2 -swf Export/flash/bin/PiratePig.swf -cp Export/flash/haxe
joshua@ponyo:~/Projects/Samples/PiratePig$
```

**Comment by jgranick** [comment]
What version of Haxe are you using?

**Comment by RudolfVonKrugstein** [comment]
I really wonder why the absolute path does not work. My haxe version is the latest stable:

```
```

**Comment by jgranick** [comment]
> There was indeed a problem which only surfaced when HAXE_STD_PATH was set (which it is by default on Linux).


Okay, looks like Simn found something in the compiler. I guess there was a chance that it could fail if HAXE_STD_PATH is set. Could you try checking if you have this environment variable, and seeing if unsetting this fixes it?

**Comment by jgranick** [comment]
I think this is resolved with Haxe 3.2, but please let me know if the environment variable helps -- if it does, then maybe I should disable the absolute paths if HAXE_STD_PATH is present. BTW, did you use the OpenFL install script for Haxe, or did you install another way?

**Comment by RudolfVonKrugstein** [comment]
I can confirm that unseting HAXE_STD_PATH works, thanks!

I installed using the openfl install script.
